### ISSUE TITLE
magithub-pull-request-new-arguments doesn't work if I have a heroku remote

### ISSUE THREAD

**Issue by aaronjensen** [vermiculus/magithub#283]
I'm not sure of the exact repro or issue, but here's what I've found. If I select "origin" from `magithub-read-repo` then `magithub-repo-remotes-for-repo` will return `("heroku" "origin")` which is incorrect.

My remotes look something like this:

```
[remote "origin"]
        url = git@github.com:org/my-app.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[remote "heroku"]
        url = https://git.heroku.com/my-app.git
```

The tests in here seem to pass:

```
               (and (string= .repo.owner.login
                             .remote.owner.login)
                    (string= .repo.name .remote.name))
```

**Comment by vermiculus** [comment]
What's the return of `magithub--url->repo` for each of those URLs?

I bet #195 is related.

**Comment by aaronjensen** [comment]
origin -> `((owner (login . "org")) (name . "my-app"))`
heroku -> `nil`

**Comment by vermiculus** [comment]
And `(magithub-repo-remotes-for-repo '((owner (login . "org")) (name . "my-app")))` returns `("heroku" "origin")`?

**Comment by vermiculus** [comment]
Due to how `magithub-repo-from-remote` works, I think I'm getting false positives. Any chance you could provide my with the real URLs?

**Comment by aaronjensen** [comment]
@vermiculus yes, it returns `("heroku" "origin")`. I emailed you the urls.

**Comment by vermiculus** [comment]
With those URLs, I was able to find the problem!  Thanks again.

Here's an updated definition for `magithub-repo-from-remote`:

``` emacs-lisp
(defun magithub-repo-from-remote (remote)
  (when-let* ((repo (magithub-repo-from-remote--sparse remote)))
    (magithub-repo repo)))
```

Ping for @aspiers and @justinweiss -- can you try out the definition above to see if this will also fix #195?

**Comment by aaronjensen** [comment]
Thanks, that seems to work for me.
